src/services/cache/index.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  LocalCacheNotAvailableError,
+  LocalCacheUndefinedError,
+  UnknownLocalCacheServiceError,
+} from "@domain/cache"
+import NodeCache from "node-cache"
+
+const localCache = new NodeCache()
+
+export const LocalCacheService = (): ILocalCacheService => {
+  const set = <T>(
+    key: CacheKeys | string,
+    value: T,
+    ttlSecs: number,
+  ): Promise<T | LocalCacheServiceError> => {
+    try {
+      const res = localCache.set<T>(key, value, ttlSecs)
+      if (res) return Promise.resolve(value)
+    } catch (err) {
+      return Promise.resolve(new UnknownLocalCacheServiceError(err))
+    }
+    return Promise.resolve(new LocalCacheNotAvailableError())
```

Comment:
For readability, these might be better closer to the `if` conditional that they're related to (at `Ln18`)